### PR TITLE
chore: ingest multiple batches to trial profiler metrics endpoint [DET-5178]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change updates POST /api/v1/trials/profiler/metrics to accept more than one batch at a time. It is crafted to return a multierror and succeed where it can and fail only where it can't. This is to avoid one malformed series causing all series to fail to be saved. This change also removes offset from the trial logs API, since it doesn't work with Elasticsearch and isn't used by anything anymore (CLI, WebUI, etc, all use ORDER_BY_DESC + limit to tail).

## Test Plan
- [x] Don't break stuff, there are tests for all of this.

## Checklist

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: master/internal/api_trials.go [hunk 1]
```diff
@@ -74,12 +74,6 @@ func (a *apiServer) TrialLogs(
 		return status.Error(codes.InvalidArgument, fmt.Sprintf("unsupported filter: %s", err))
 	}
 
-	total, err := a.m.trialLogBackend.TrialLogsCount(int(req.TrialId), filters)
-	if err != nil {
-		return fmt.Errorf("failed to get trial count from backend: %w", err)
-	}
-	offset, limit := api.EffectiveOffsetNLimit(int(req.Offset), int(req.Limit), total)
```
Comment (stoksc) [inline master/internal/api_trials.go:81]: We don't calculate effective limit any, which was just min(log_count, limit), but this is fine; if limit is less, this is the same behavior, if it is more, the `BatchStreamProcessor` will terminate when it gets a nil or len(batch) == 0 batch.

**Comment by stoksc** [comment]
These touch a bunch of the same files so I just shoved them into a PR. The two commit are almost separated right except the tests 🤦 (forgot I changed it in both commits). If this is an issue lmk, but it still should be easily reviewable.

**Review by dzhu — APPROVED** [review]

File: master/internal/api/stream.go [hunk 2]
```diff
@@ -88,43 +88,46 @@ func NewBatchStreamProcessor(
...
+			case !p.req.Follow && p.req.Limit <= 0:
+				return nil
+			}
+		}
+
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-t.C:
+			continue
```
[the remark this answers is not in the document]
Comment (stoksc) [reply master/internal/api/stream.go:130]: Yeah, I just left it to clarify it was intended.

File: master/internal/api/stream.go [hunk 3]
```diff
@@ -88,43 +88,46 @@ func NewBatchStreamProcessor(
...
+			p.req.Offset += batch.Size()
+			switch err := p.process(batch); {
+			case err != nil:
+				return fmt.Errorf("failed while processing batch: %w", err)
+			case !p.req.Follow && p.req.Limit <= 0:
+				return nil
+			}
+		}
+
+		select {
```
[the remark this answers is not in the document]
Comment (stoksc) [reply master/internal/api/stream.go:126]: If it happens, any time other than the first iteration it's the same as always, on the first, 🤷. I could check it to start once too. This change was due to how `time.Ticker` sends the first tick after the tick period.

[the remark this answers is not in the document]
Comment (stoksc) [reply master/internal/api/stream.go:126]: Unless the request is canceled immediately, we wouldn't waste any time fetching, and even if it, we'll at least not send the result on the dead ctx still.

**Pull request merged**